include/OpenSpaceToolkit/Astrodynamics/Trajectory/State.hpp

```diff
@@ -83,6 +83,19 @@ class State
 
     State(const Instant& anInstant, const Position& aPosition, const Velocity& aVelocity);
 
+    /// @brief                  Copy constructor.
+    ///
+    /// @param                  [in] aState An existing State
+
+    State(const State& aState);
+
+    /// @brief                  Copy-assignment operator
```

Comment:
Why are we adding this? 🤔 Doesn't seem related to the Meta-class?

Reply:
The copy constructor is needed here:
https://github.com/open-space-collective/open-space-toolkit-astrodynamics/pull/244/files#diff-f995720c611c0d79a20e7b908e36d9360719bb3784955f510982c459db799006R32

When sub-classing a `State`, we need to call it's `__init__` during construction of the sub-class. In this case, I don't want the `State` object to actually be produced using the normal `State` constructor, but via a `StateBuilder`.

There could be a way around this by ignoring the `StateBuilder`, but I think it's a good idea to use it so we never have to re-code all of the extra stuff inside of it.


The copy assignment operator is because after creating the copy constructor, the compiler was giving warnings that it could not use the default copy operator anymore, so I added an explicit one.